src/CMakeLists.txt

```diff
@@ -164,7 +164,7 @@ if(VM_UNCHECKED)
   set(LEAN_EXTRA_CXX_FLAGS "${LEAN_EXTRA_CXX_FLAGS} -D LEAN_VM_UNCHECKED")
 endif()
 
-if(AUTO_THREAD_FINALIZATION)
+if(AUTO_THREAD_FINALIZATION AND NOT MSVC)
   set(LEAN_EXTRA_CXX_FLAGS "${LEAN_EXTRA_CXX_FLAGS} -D LEAN_AUTO_THREAD_FINALIZATION")
```

Comment:
Why did you have to disable this option when using VS?
BTW, this option is relevant when creating the lean shared object (aka DLL on Windows). 
This DLL is not essential. It is only relevant for users that want to embed Lean into their applications.

Reply:
AFAIU, when AUTO_THREAD_FINALIZATION is disabled, the code fallbacks to using __thread variables instead of using pthreads. That code was already written to be compatible with MSVC and compiles fine.
So nothing is disabled AFAICT; it's just it's using a non-pthread implementation.  I think the name of the cmake option is a bit misleading.